### PR TITLE
improved testing and coverage with sigstore package mock

### PR DESCRIPTION
**Pull request by rodrigolc**
**Pull Request check list**

- [ ] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

**Description of change**

**Which issue this PR fixes**

### CODE REVIEW

**Review by willallves — APPROVED** [review]

File: pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore.go [hunk 1]
```diff
@@ -41,35 +41,37 @@ func (sigstore Sigstoreimpl) FetchSignaturePayload(imageName string, rekorURL st
...
-	// TODO: decide if http is allowed
-	if rekorURI.Scheme != "" && rekorURI.Scheme != "http" && rekorURI.Scheme != "https" {
-		return nil, errors.New("Invalid rekor URL Scheme: " + rekorURI.Scheme)
+	if rekorURL != "" {
+		rekorURI, err := url.Parse(rekorURL)
+		if err != nil {
+			message := fmt.Sprint("Error parsing rekor URI: ", err.Error())
+			return nil, errors.New(message)
+		}
+		// TODO: decide if http is allowed
```
[the remark this answers is not in the document]
Comment (rodrigolc) [reply pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore.go:51]: Good catch, it wasn't. HTTP doesn't make sense in a zero-trust environment, I'll change it and restrict it to HTTPS, if any.

File: pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore.go [hunk 2]
```diff
@@ -148,8 +150,9 @@ func getImageSubject(verified []oci.Signature) string {
 		outputKeys = append(outputKeys, ss)
 	}
 	b, err := json.Marshal(outputKeys)
+	// shouldn't ever happen, since any garbled text would be caught by the unmarshal earlier
 	if err != nil {
-		log.Println("Error generating the output:", err.Error())
+		log.Println("Error generating signature: ", err.Error())
 		return ""
 	}
```
[the remark this answers is not in the document]
Comment (rodrigolc) [reply pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore.go:157]: Fair enough. I'll remove this bit.

File: pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore_test.go [hunk 3]
```diff
@@ -289,6 +538,69 @@ func Test_getImageSubject(t *testing.T) {
 		want string
 	}{
 		// TODO: Add test cases.
```
Comment (willallves) [inline pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore_test.go:540]: is it still valid?

Comment (rodrigolc) [reply pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore_test.go]: I've removed the TODO's

**Review by tjamir — APPROVED** [review]

**Pull request merged**